include/spinlock.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2018 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef _SPINLOCK_H
+#define _SPINLOCK_H
+
+#include <atomic.h>
+
+typedef struct k_spinlock_key {
+	int key;
+} k_spinlock_key_t;
+
+struct k_spinlock_t {
```

Comment:
this is a weird name.
stuff that ends in _t is typically typedef'd.
can we just call it "struct k_spinlock"?